lib/compress.js

```diff
@@ -4655,4 +4655,41 @@ merge(Compressor.prototype, {
     OPT(AST_PrefixedTemplateString, function(self, compressor){
         return self;
     });
+
+    OPT(AST_ObjectKeyVal, function(self, compressor){
+        // p:function(){} ---> p(){}
+        // p:function*(){} ---> *p(){}
+        // p:async function(){} ---> async p(){}
+        // p:()=>{} ---> p(){}
+        // p:async()=>{} ---> async p(){}
+        if (compressor.option("ecma") >= 6) {
+            var key = self.key;
+            var value = self.value;
+            var is_arrow_with_block = value instanceof AST_Arrow
+                && Array.isArray(value.body)
```

Comment:
I forgot to add tests for excluding arrow expressions from being transformed to concise methods (since arrow form is shorter):
```
$ echo 'console.log(({ a: () => 3 }).a());' | bin/uglifyjs -c ecma=6
console.log({a:()=>3}.a());

$ echo 'console.log(({ a: () => {return 3;} }).a());' | bin/uglifyjs -c ecma=6
console.log({a:()=>3}.a());

$ echo 'console.log(({ a(){return 3;} }).a());' | bin/uglifyjs -c ecma=6
console.log({a(){return 3}}.a());
```